sklearn/tests/test_docstring_parameters.py

```diff
@@ -331,21 +331,46 @@ def _get_all_fitted_attributes(estimator):
 @skip_if_no_numpydoc
 def test_precision_recall_f_score_docstring_consistency():
     """Check docstrings parameters of related metrics are consistent."""
+    metrics_to_check = [
+        metrics.precision_recall_fscore_support,
+        metrics.f1_score,
+        metrics.fbeta_score,
+        metrics.precision_score,
+        metrics.recall_score,
+    ]
     assert_docstring_consistency(
-        [
-            metrics.precision_recall_fscore_support,
-            metrics.f1_score,
-            metrics.fbeta_score,
-            metrics.precision_score,
-            metrics.recall_score,
-        ],
+        metrics_to_check,
         include_params=True,
-        # "average" - in `recall_score` we have an additional line: 'Weighted recall
-        # is equal to accuracy.'.
         # "zero_division" - the reason for zero division differs between f scores,
         # precison and recall.
         exclude_params=["average", "zero_division"],
     )
+    description_regex = r"""This parameter is required for multiclass/multilabel targets\.
+        If ``None``, the metrics for each class are returned\. Otherwise, this
+        determines the type of averaging performed on the data:
+        ``'binary'``:
+            Only report results for the class specified by ``pos_label``\.
+            This is applicable only if targets \(``y_\{true,pred\}``\) are binary\.
+        ``'micro'``:
+            Calculate metrics globally by counting the total true positives,
+            false negatives and false positives\.
+        ``'macro'``:
+            Calculate metrics for each label, and find their unweighted
+            mean\.  This does not take label imbalance into account\.
+        ``'weighted'``:
+            Calculate metrics for each label, and find their average weighted
+            by support \(the number of true instances for each label\)\. This
+            alters 'macro' to account for label imbalance; it can result in an
+            F-score that is not between precision and recall\.[\s\w]*\.*
+        ``'samples'``:
+            Calculate metrics for each instance, and find their average \(only
+            meaningful for multilabel classification where this differs from
+            :func:`accuracy_score`\)\."""  # noqa E501
```

Comment:
for the purpose of the test, we could maybe pass a real regex pattern instead. For instance, we could require to only match the type of average, e.g., 'binary', 'micro', etc.

Reply:
In this case, we could also parametrize the test by passing or not the `description_regex` parameter.

Reply:
>  we could maybe pass a real regex pattern instead. For instance, we could require to only match the type of average, e.g., 'binary', 'micro', etc.

Sorry I don't follow. Do you mean take only specific average types, e.g., 'binary', 'micro', and match the extracted description between objects?
That would be another way to do it, but the current regex pattern just matches descriptions from all objects against the given regex...